### PR TITLE
feat: add transition for dark/light theme change

### PR DESCRIPTION
**Pull request by ChristianBusshoff**
- Add theme change transition
- Info: .transition-active required the !important attribute, otherwise it didn't work.

### CODE REVIEW

File: apps/docs/src/development/theming.md [hunk 1]
```diff
@@ -68,3 +68,17 @@ const { store: colorScheme } = useColorMode();
 ```
 
 Alternatively, you can use the [OnyxColorSchemeDialog](https://storybook.onyx.schwarz/?path=/docs/support-colorschemedialog--docs) component to build your own custom component.
+
+### Transistion
```
Comment (larsrickert) [inline apps/docs/src/development/theming.md:72]: I would document this per default, so maybe we can include this into the docs above that documents `useColorMode()`

File: packages/sit-onyx/src/composables/themeTransition.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  //onMounted because of Server-Side-Rendering
+  onMounted(() => {
+    watch([theme], () => {
+      const root = document.documentElement;
+
+      const handleTransitionEnd = () => {
+        root.classList.remove("onyx-transition-active");
+      };
+      root.addEventListener("transitionend", handleTransitionEnd, { once: true });
+      root.classList.add("onyx-transition-active");
```
Comment (larsrickert) [inline packages/sit-onyx/src/composables/themeTransition.ts:14]:
```suggestion
     const className = "onyx-transition-active";

      const handleTransitionEnd = () => {
        root.classList.remove(className);
      };
      root.addEventListener("transitionend", handleTransitionEnd, { once: true });
      root.classList.add(className);
```

File: apps/docs/src/development/theming.md (hunk 1 above)
Comment (larsrickert) [inline apps/docs/src/development/theming.md:72]: Please also create a changset for the new ``useThemeTransistion` composable + link the docs page :)

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/composables/themeTransition.ts (hunk 2 above)
Comment (ChristianBusshoff) [reply packages/sit-onyx/src/composables/themeTransition.ts]: changed it

File: apps/docs/src/development/theming.md (hunk 1 above)
Comment (ChristianBusshoff) [reply apps/docs/src/development/theming.md]: changed it

File: .changeset/tame-frogs-pump.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+"sit-onyx": minor
+"docs": minor
+---
+
+- added light/dark mode transition
+- created the useThemeTransition composable, which observes changes between light and dark mode and dynamically adds the onyx-transition-active class during the transition for a smooth visual effect.
```
Comment (larsrickert) [inline .changeset/tame-frogs-pump.md:7]:
```suggestion
- created the useThemeTransition composable, which observes changes between light and dark mode and dynamically adds the onyx-transition-active class during the transition for a smooth visual effect. See our [docs](https://onyx.schwarz/development/typography.html) on how to use it.
```

File: apps/docs/src/development/theming.md
Comment (larsrickert) [inline apps/docs/src/development/theming.md:1]: Did you remove this part intentionally? I would still like to keep it:

```md
Alternatively, you can use the [OnyxColorSchemeDialog](https://storybook.onyx.schwarz/?path=/docs/support-colorschemedialog--docs) component to build your own custom component.
```

**Review by larsrickert — APPROVED** [review]
Looks good to me :) @JoCa96 Would be interested in your opinion on this approach if you have time

File: apps/docs/src/development/theming.md
Comment (ChristianBusshoff) [reply apps/docs/src/development/theming.md:1]: changed it

File: .changeset/tame-frogs-pump.md (hunk 3 above)
Comment (ChristianBusshoff) [reply .changeset/tame-frogs-pump.md:7]: changed it

**Pull request merged**